components/LanguageSwitcher.tsx

```diff
@@ -3,13 +3,18 @@ import { useRouter } from 'next/router'
 
 import { useTranslation } from 'next-i18next'
 
-export const LanguageSwitcher: React.FC = () => {
+export interface LanguageSwitcherProps {
+  mobile: boolean
+}
+
+export const LanguageSwitcher: React.FC<LanguageSwitcherProps> = ({ mobile = false }) => {
   const { t } = useTranslation('common')
   const router = useRouter()
 
+  const queryString = mobile ? { from: 'uiom' } : {}
   return (
     <div className="language-switcher">
-      <Link href="/" locale={router.locale === 'en' ? 'es' : 'en'}>
+      <Link href={{ pathname: '/', query: queryString }} locale={router.locale === 'en' ? 'es' : 'en'}>
```

Comment:
`href` uses the [URL Object syntax](https://nextjs.org/docs/api-reference/next/link#with-url-object)